### PR TITLE
Add health checks

### PR DESCRIPTION
**Pull request by danihodovic**
Adds health checks to the application so that we can monitor it's state.
This is important for two reasons:

- On deployment the deployer pings this endpoint to find out if the
  application has booted correctly.

- Once we introduce load balancers the load balancer can detect if the
  application is unhealthy and direct traffic to
  another instance.

### CODE REVIEW

**Comment by danihodovic** [comment]
Do not merge until I can confirm that this works on a testing server. I need a manual confirmation due to the fact that we use ganache in tests and ganache doesn't provide a way of health checking. The `isConnected()` method does not exist, so we stub it into tests.

**Comment by danihodovic** [comment]
```
(node:26) UnhandledPromiseRejectionWarning: TypeError: web3.isConnected is not a function
    at _callee$ (/app/dist/routes/health_check.js:42:24)
    at tryCatch (/app/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:62:40)
```

It turns out that the method that is in the docs doesn't exist https://github.com/ethereum/wiki/wiki/JavaScript-API#web3isconnected. This is understandable because  API doc describe 0.2.x.

I didn't find an `isConnected()` method in the source of `1.0.0-beta.34` so I'll look for a workaround.

**Comment by danihodovic** [comment]
I ended up using `web3.eth.getNodeInfo()` for the eth health check. I ran a test in development where it uses the actual node (not ganache) and disconnected it from the eth node.

It seems to work.
```
curl -i localhost:82/health_check
HTTP/1.1 500 Internal Server Error

{"msg":"Failed to call getNodeInfo"}
```

Mongo health checks are covered in tests.

Health checks for a healthy node also work in a real environment.
```
curl -i localhost:82/health_check
HTTP/1.1 200 OK
```

File: test/routes/health_check.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    expect(response.status).to.eql(500);
+    expect(response.body).to.eql({msg: 'Is not connected to MongoDB'});
+  });
+
+  it('returns 500 if web3 is not connected', async () => {
+    // We can't close the connection to web3 because we're using ganache,
+    // mocking is the second best option.
+    const stub = sinon.stub(apparatus.web3.eth, 'getNodeInfo');
+    stub.returns(Promise.reject('Error: Invalid JSON RPC response: ""'));
```
Comment (marekkirejczyk) [inline test/routes/health_check.js]: I think stub needs to be released.

File: src/server.js [hunk 2]
```diff
@@ -17,9 +17,11 @@ import bundlesRouter from './routes/bundles';
 import eventsRouter from './routes/events';
 import tokenRouter from './routes/token';
 import nodeInfoRouter from './routes/nodeinfo';
+import healthCheckHandler from './routes/health_check';
 
 export default class Server {
-  constructor(modelEngine, config) {
+  constructor(modelEngine, mongoClient, config) {
```
Comment (marekkirejczyk) [inline src/server.js]: I would argue that `mongoClient` should be taken from `modelEngine`.

File: src/server.js [hunk 3]
```diff
@@ -40,6 +42,7 @@ export default class Server {
     app.use('/events', eventsRouter(this.modelEngine.tokenAuthenticator, this.modelEngine.identityManager, this.modelEngine));
     app.use('/token', tokenRouter(this.modelEngine.tokenAuthenticator, this.config));
     app.use('/bundle', bundlesRouter(this.modelEngine));
+    app.use('/health_check', healthCheckHandler(this.mongoClient, this.modelEngine.proofRepository.web3));
```
Comment (marekkirejczyk) [inline src/server.js]: Great idea about health check. I would argue to use RESTfull naming (i.e. simply `/health`).

File: src/routes/health_check.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  try {
+    await web3.eth.getNodeInfo();
+  } catch (err) {
+    res.status(500)
+      .type('json')
+      .send({msg: 'Failed to call getNodeInfo'});
+    return;
+  }
```
Comment (marekkirejczyk) [inline src/routes/health_check.js]: Do we want to return 500 or json with detailed information, e.g.:
```js
{
mongo: {isConnected: false}, 
nodeInfo: {
  ...
}
}
```

**Pull request merged**